src/util.cpp

```diff
@@ -570,16 +579,19 @@ void ArgsManager::ParseParameters(int argc, const char* const argv[])
         if (key.length() > 1 && key[1] == '-')
             key.erase(0, 1);
 
-        // Transform -nofoo to -foo=0
-        InterpretNegatedOption(key, val);
-
-        m_override_args[key].push_back(val);
+        // Check for -nofoo
+        if (InterpretNegatedOption(key, val)) {
+            m_override_args[key].clear();
```

Comment:
Shouldn't a negated command line argument clear m_config_args too? As an example, if you had `arg=a` `arg=b` in config the and `-arg=c -noarg -noarg=d`, it would seem weird for the result to be `[a b d]` instead of `[d]`.

Reply:
ParseParameters is always called prior to ReadConfigFile. Might be worth noting somewhere that you have to do things that way (though it isn't really a choice: if you specify the config file via `-conf`, you don't know what file to pass to ReadConfigFile until you've called ParseParameters).

Reply:
https://github.com/bitcoin/bitcoin/pull/11862#discussion_r182556323

> ParseParameters is always called prior to ReadConfigFile

So this means the change I suggested wouldn't actually fix the bug. But you agree this is  introducing a bug that wasn't in @eklitzke's initial implementation from #12713?

Reply:
> But you agree this is introducing a bug

Never mind, this bug is actually present in #12713 too because of the initialization order. I guess I thought negated option tracking would provide a clean way to override config file list options instead of just appending to them, but I guess this isn't the way it works.